WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2909,16 +2916,36 @@ static GSourceFuncs _handlerIntervention =
                     webkit_memory_pressure_settings_free(memoryPressureSettings);
                 }
 #endif
+
+#if WEBKIT_CHECK_VERSION(2, 42, 0)
+                double originStorageRatio = -1.0;    // -1.0 means WebKit will use the default quota (1GB)
+                if (_config.IndexedDBSize.IsSet() && _config.IndexedDBSize.Value() != 0) {
+                    originStorageRatio = static_cast<double>(_config.IndexedDBSize.Value());
+                }
+
+                double totalStorageRatio = -1.0;    // -1.0 means there's no limit for the total storage
+                if (_config.IndexedDBTotalSize.IsSet() && _config.IndexedDBTotalSize.Value() != 0) {
+                    totalStorageRatio = static_cast<double>(_config.IndexedDBTotalSize.Value());
+                }
+
                 auto* websiteDataManager = webkit_website_data_manager_new(
                     "local-storage-directory", wpeStoragePath,
                     "disk-cache-directory", wpeDiskCachePath,
                     "local-storage-quota", localStorageDatabaseQuotaInBytes,
                     "indexeddb-directory", indexedDBPath,
-                    "per-origin-storage-quota", indexedDBSizeBytes,
+                    "origin-storage-ratio", originStorageRatio,
+                    "total-storage-ratio", totalStorageRatio,
+                     nullptr);
+                g_free(indexedDBPath);
+#else
+                auto* websiteDataManager = webkit_website_data_manager_new(
+                    "local-storage-directory", wpeStoragePath,
+                    "disk-cache-directory", wpeDiskCachePath,
```

Comment:
If I followed the discussion correctly the outcome is that with  IndexexDBSize we do not have to be backwards compatible. Just to make sure do older webkit versions have a sensible default as it is now not set at all or do we have to do that explicitly here?

Reply:
Indeed the idea is that we won't be backwards compatible with `IndexedDB`, since nobody was using this feature so far anyway. `IndexedDB` is disabled by default, so in our case it will be always disabled for any older WebKit versions, and since 2.42 users can enabled it in the config. Is this correct @modeveci?